pkg/auth/authorizer/abac/abac.go

```diff
@@ -52,7 +53,46 @@ func (p policyLoadError) Error() string {
 type policyList []*api.Policy
 
 // TODO: Have policies be created via an API call and stored in REST storage.
-func NewFromFile(path string) (policyList, error) {
+func NewFromFile(path string) (*policyList, error) {
+
+	pl, err := newFromFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an empty watcher
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch the policy file
+	err = watcher.Add(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// On write operations on the policy file, reload policy file
+	go func() {
+		for {
+			select {
+			case ev := <-watcher.Events:
+				switch ev.Op {
+				case fsnotify.Write:
+					_, err = newFromFile(path, pl)
```

Comment:
load, handle errors, then swap rules... don't overwrite rules until we're sure we loaded the file correctly